components/backup_text/src/bin/rewrite/main.rs

```diff
@@ -151,22 +162,38 @@ async fn worker(opt: Opt) -> Result<()> {
         .unwrap();
     info!("rewrite data files all done");
 
-    let new_meta = {
-        let mut meta = meta;
-        mutate_data_files(&storage, &new_storage, &mut meta, |file| {
-            let mutated_file = mutated_file_map
-                .get(file.get_name())
-                .cloned()
-                .ok_or_else(|| anyhow!("file not mutated: {}", file.get_name()))
-                .unwrap();
-            *file = mutated_file;
-        })
-        .await?;
-        meta
-    };
-
-    write_message(&new_storage, META_FILE, new_meta)?;
-    info!("update meta file done");
+    // Post work
+    match mode {
+        RewriteMode::ToText | RewriteMode::ToSst => {
+            // update meta file for correct restoration
+            let new_meta = {
+                let mut meta = meta;
+                mutate_data_files(&storage, &new_storage, &mut meta, |file| {
+                    let mutated_file = mutated_file_map
+                        .get(file.get_name())
+                        .cloned()
+                        .ok_or_else(|| anyhow!("file not mutated: {}", file.get_name()))
+                        .unwrap();
+                    *file = mutated_file;
+                })
+                .await?;
+                meta
+            };
+            write_message(&new_storage, META_FILE, new_meta)?;
+            info!("update meta file done");
+        }
+        RewriteMode::ToCsv => {
+            // cleanup
+            for file in mutated_file_map.into_values() {
+                let mut path = new_path.clone();
+                path.push(file.name);
+                if fs::metadata(&path)?.len() == 0 {
+                    fs::remove_file(&path)?;
+                }
+            }
+            info!("cleanup done");
```

Comment:
We can clean up empty files after `TextWriter::finish` through `TextWriter::get_size` and `TextWriter::cleanup`